src/coreclr/vm/cdacdata.h

```diff
@@ -7,9 +7,10 @@
 // See datadescriptor.h
 //
 // This struct enables exposing information that is private to a class to the cDAC. For example,
-// if class C has private information that must be provided, declare cdac_data<T> as a friend:
+// if class C has private information that must be provided, declare cdac_data<T> as a friend of C
+// where T is the specialization of cdac_data that will expose the information. For example:
 //
-//     template<typename T> friend struct ::cdac_data;
+//     friend struct ::cdac_data<T>;
```

Comment:
nit1: `T` is usually a type parameter name, so this might be a bit confusing to read.
Use `D` or `Class2` or something.

nit2: maybe useful to mention that in the common case `Class2` will be `C` itself.

maybe expand the example below so it is complete:

```cxx
  class C {
    private:
       int F;
       friend struct ::cdac_data<C>;
  };
  template<> struct cdac_data<C> {
    static constexpr size_t F_Offset = offsetof(C, F);
  };
```

Reply:
Expanded example and added note that in the common case type `D` will be type `C`.